### ISSUE TITLE
Result.assumptions always returns empty tuple iterator

### ISSUE THREAD

**Issue by Jus-Codin** [jaraco/wolframalpha#29]
Assumptions seem to be inaccessible from `Result.assumptions`, which always returns an empty tuple iterator
This seems to be caused by the function [here](https://github.com/jaraco/wolframalpha/blob/main/wolframalpha/__init__.py#L189).

When `Result.assumptions` is called, the function is attempting to access the key `assumption`

However, based on the sample query `pi`, assumptions are stored in the key `assumptions`:
```
{'@datatypes': '',
 '@error': 'false',
 ...
 'assumptions': {'@count': '1',
                 'assumption': {'@count': '6',
                                '@template': 'Assuming "${word}" is ${desc1}. '
                                             'Use as ${desc2} instead',
                                '@type': 'Clash',
                                '@word': 'pi',
                                'value': [...]
                 }
 },
 ...
}
```
Is this a bug, or intended behaviour?

**Comment by jaraco** [comment]
I do see that the code [explicitly mentions assumptions](https://github.com/jaraco/wolframalpha/blob/9cbd7666a71eadcc05cf4d8b354a6e8c3959ede3/wolframalpha/__init__.py#L256) (plural) and later [strips the `s`](https://github.com/jaraco/wolframalpha/blob/9cbd7666a71eadcc05cf4d8b354a6e8c3959ede3/wolframalpha/__init__.py#L192) to get the singular form. However, tracing the logic, I don't see how it could ever find assumptions. And looking at [this test](https://github.com/jaraco/wolframalpha/blob/9cbd7666a71eadcc05cf4d8b354a6e8c3959ede3/tests/test_client.py#L40-L41), it only performs some pretty weak assertions on the assumptions (it won't catch the condition you're describing). So yeah, probably a bug.

**Comment by jaraco** [comment]
Fix released in v5.0.1.
